dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -457,499 +401,249 @@ public void checkInvalidActionTest()
+     */
+    @Test
+    public void does_a_site_specific_vanity_override_system_host_vanity()
+            throws Exception {
+
+        DefaultVanityUrl siteVanity = null;
+        DefaultVanityUrl systemHostVanity = null;
+
+
+
+        final long currentTime = System.currentTimeMillis();
+        final String uri = "/" + currentTime + "_testing" + "(.*)";
+        final String requestedURL = "/" + currentTime + "_testing" + "RANDOM_CHARACTERS";
+        final String nonExistingURL = "/nonexisting/should404/index" + currentTime;
+
+        //------------------------------------
+        //Create a VanityURL for the default host
+        //------------------------------------
+        siteVanity = filtersUtil
+                .createVanityUrl("test Vanity Url " + System.currentTimeMillis(),
+                        defaultHost.getIdentifier(),
+                        uri, "https://www.google.com", 200, 1, defaultLanguage.getId());
+        filtersUtil.publishVanityUrl(siteVanity);
+        
+        // nothing in cache
+        List<CachedVanityUrl> siteCache= cache.getSiteMappings(defaultHost, defaultLanguage);
+        assert( siteCache==null);
+        
+        //None of these exist in cache
+        Optional<CachedVanityUrl> vanityURLCached = cache.getDirectMapping(uri, defaultHost, defaultLanguage);
+        assert(vanityURLCached ==null );
+        vanityURLCached = cache.getDirectMapping(requestedURL, defaultHost, defaultLanguage);
+        assert(vanityURLCached ==null );
+        vanityURLCached = cache.getDirectMapping(nonExistingURL, defaultHost, defaultLanguage);
+        assert(vanityURLCached ==null );
+
+        
+        // one hit loads cache
+        vanityURLCached= vanityUrlAPI.resolveVanityUrl(requestedURL, defaultHost, defaultLanguage);
+        assert( vanityURLCached.isPresent());
+        
+        siteCache= cache.getSiteMappings(defaultHost, defaultLanguage);
+        
+        // it lives in the host cache
+        assert( siteCache!=null);
+        assert(siteCache.contains(vanityURLCached.get()));
+        Optional<CachedVanityUrl> directCached = cache.getDirectMapping(requestedURL, defaultHost, defaultLanguage);
+        // and the direct cache
+        assertEquals(vanityURLCached.get(), directCached.get());
+        
+        
+
+        //------------------------------------
+        //Create the VanityURL for the SYSTEM_HOST
+        //------------------------------------
+
+        systemHostVanity = filtersUtil.createVanityUrl("test Vanity Url " + System.currentTimeMillis(), Host.SYSTEM_HOST, uri,
+                        "https://www.google.com", 200, 1, defaultLanguage.getId());
+        filtersUtil.publishVanityUrl(systemHostVanity);
+
+        //Should not exist in direct cache (the save/publish wipes cache) or in the host cache
+        vanityURLCached = cache.getDirectMapping(requestedURL, defaultHost, defaultLanguage);
+        Assert.assertNull(vanityURLCached);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertNull' due to existing static import 'org.junit.Assert.assertNull'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)